src/odin_devices/bme280.py

```diff
@@ -79,24 +82,32 @@
                        STANDBY_TC_62_5, STANDBY_TC_125, STANDBY_TC_250,
                        STANDBY_TC_500, STANDBY_TC_1000)
 
-class BME280(SPIDevice):
+class BME280():
     """BME280 class.
 
-    This class implements support for the BME280 device.
+    This class implements support for the BME280 device, for SPI and I2C.
     """
-    def __init__(self):
+    def __init__(self, use_spi=True, i2c_busnum=0):
         """Initialise the BME280 device.
 
-        Any SPI settings can be adjusted with the functions in spi_device.
-        SPIDevice.__init__ is provided bus, device, bits_per_word (optional) and hz (optional).
-        """
-        SPIDevice.__init__(self, 0, 0)  # bus, device
-
-        # This device is compatible with SPI modes 0 and 3 (00, 11).
-        self.set_mode(0)
+        :param use_spi: bool to specify device type for bme to use. 
+        Default True = SPI will be used, False = I2C.
+        :param i2c_busnum: bus number for i2c device. only used when use_spi is false. Default: 0
 
-        # Longest transaction is 24 bytes read, one written.
-        self.set_buffer_length(25)
+        Device settings can be adjusted with the functions in the respective device class.
+        """
+        self.use_spi = use_spi
+
+        if self.use_spi:  # using SPI
+            self.device = SPIDevice(bus=0, device=0)
```

Comment:
Reviewing this PR (nice work BTW!) had made me realise that sometimes we need to be able to override the bus and device numbers for an SPI device. For instance, on the LOKI board the external SPI bus = 1. So I suggest that the `__init__` method takes three args: `use_spi` as implemented, `bus` replaces `i2c_busnum` and can be used for both I2C and SPI (with a default of 0) and `device` can be used for SPI (with a default of 0)